src/proj7BittingCerratoCohenEllmer/bantam/lexer/precusortokens/PrecursorTokenFactory.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * File: PrecursorTokenFactory.java
+ * Author: cbitting
+ * Date: 4/8/2021
+ */
+package proj7BittingCerratoCohenEllmer.bantam.lexer.precusortokens;
+
+import java.util.Stack;
+
+/**
+ * This class generates precursor tokens based on a passed character.
+ */
+public class PrecursorTokenFactory {
```

Comment:
Did u read the design pattern book?

I know a factory pattern exists but Im pretty unaware.

I was hoping to learn about design patterns in this class and so far we lightly went over MVC... maybe I should just read more...